src/TraceEvent/Computers/TraceManagedProcess.cs

```diff
@@ -1877,6 +1877,10 @@ public void EnsureBGCRevisitInfoAlloc()
         /// </summary>
         public int HeapCount { get; private set; } = -1;
         /// <summary>
+        /// Number of Generations (0,1,2,LOH,[POH]), which is either 4 or 5 depending on version
+        /// </summary>
+        public int GenerationCount { get { return this.PerHeapHistories?.FirstOrDefault()?.GenData.Length ?? 0; } }
```

Comment:
Do we need to check `.GenData` for `null` as well?
```suggestion
        public int GenerationCount { get { return this.PerHeapHistories?.FirstOrDefault()?.GenData?.Length ?? 0; } }
```

Reply:
Yes, I think this would be a good change.